spec/index.bs

```diff
@@ -1894,6 +1900,111 @@ dictionary GPUVertexAttributeDescriptor {
 };
 </script>
 
+## {{GPUPipelineBase/getBindGroupLayout()|getBindGroupLayout}} ## {#get-bind-group-layout}
+
+<dl dfn-type="method" dfn-for="GPUPipelineBase">
+    : <dfn>getBindGroupLayout(unsigned long index)</dfn>
+    ::
+        <div algorithm="GPUProgrammablePipeline.getBindGroupLayout(unsigned long index)">
+            1. If |index| is greater or equal to {{GPULimits/maxBindGroups}} :
+
+                1. Throw a {{RangeError}}.
+
+            1. If |this|.getBindGroupLayout has already been called with |index|:
+
+                1. Return the same object.
+
+            1. If |this| is an error:
+
+                1. Return an error {{GPUBindGroupLayout}}.
+
+            1. Let |bindings| be an empty sequence.
+            1. For each {{GPUProgrammableStageDescriptor}} |stage| in the descriptor for |this|:
+
+                1. For each resource |resource| in |stage|.{{GPUProgrammableStageDescriptor/module}}
+                    for the |stage|.{{GPUProgrammableStageDescriptor/entryPoint}} entry-point:
+
+                    1. If |resource|'s "group" decoration is not equal to |index|:
+
+                        1. Continue to the next loop iteration over resources.
+
+                    1. Let |binding| be a new {{GPUBindGroupLayoutBinding}}.
+                    1. Set |binding|.{{GPUBindGroupLayoutBinding/binding}} to |resource|'s "binding"
+                        decoration.
+                    1. Set |binding|.{{GPUBindGroupLayoutBinding/visibility}} to
+                        {{GPUShaderStage/VERTEX}} | {{GPUShaderStage/FRAGMENT}} |
+                        {{GPUShaderStage/COMPUTE}}.
+
+                        Issue: Have a set of all valid {{GPUShaderStage}} on the device.
+
+                    1. If |resource| is for a sampler binding:
+
+                            1. Set |binding|.{{GPUBindGroupLayoutBinding/type}} to
+                                {{GPUBindingType/sampler}}.
+
+                    1. If |resource| is for a buffer binding:
+
+                        1. If |resource| is for a uniform buffer:
+
+                            1. Set |binding|.{{GPUBindGroupLayoutBinding/type}} to
+                                {{GPUBindingType/uniform-buffer}}.
+
+                        1. If |resource| is for a storage buffer:
+
+                            1. If |resource| is read-only:
+
+                                1. Set |binding|.{{GPUBindGroupLayoutBinding/type}} to
+                                    {{GPUBindingType/readonly-storage-buffer}}.
+
+                            1. Else:
+
+                                1. Set |binding|.{{GPUBindGroupLayoutBinding/type}} to
+                                    {{GPUBindingType/storage-buffer}}.
+                                1. Set |binding|.{{GPUBindGroupLayoutBinding/visibility}} to
```

Comment:
Related to my previous question. Also, we know if the pipeline is compute one versus the render one, so `COMPUTE` is trivially included or eliminated.